dspace-server-webapp/src/test/java/org/dspace/app/rest/VersionRestRepositoryIT.java

```diff
@@ -202,4 +252,1186 @@ public void versionItemTestWrongId() throws Exception {
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$.inArchive", Matchers.is(true)));
+
+        // item that linked last version is not archived
+        getClient(adminToken).perform(get("/api/core/items/" + lastVersionItem.getID()))
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$.inArchive", Matchers.is(false)));
+
+        // if there is item not archived, we can not create new version
+        getClient(adminToken).perform(post("/api/versioning/versions")
+                             .param("summary", "check first version")
+                             .contentType(MediaType.parseMediaType(RestMediaTypes.TEXT_URI_LIST_VALUE))
+                             .content("/api/core/items/" + item.getID()))
+                             .andExpect(status().isUnprocessableEntity());
+
+        // retrieve the workspace item
+        getClient(adminToken).perform(get("/api/submission/workspaceitems/search/item")
+                             .param("uuid", String.valueOf(lastVersionItem.getID())))
+                             .andExpect(status().isOk())
+                             .andDo(result -> idRef.set(read(result.getResponse().getContentAsString(), "$.id")));
+
+        // submit the workspaceitem to complete the deposit
+        getClient(adminToken).perform(post(BASE_REST_SERVER_URL + "/api/workflow/workflowitems")
+                             .content("/api/submission/workspaceitems/" + idRef.get())
+                             .contentType(textUriContentType))
+                             .andExpect(status().isCreated());
+
+        // now the item is archived
+        getClient(adminToken).perform(get("/api/core/items/" + lastVersionItem.getID()))
+                             .andExpect(status().isOk())
+                             .andExpect(jsonPath("$.inArchive", Matchers.is(true)));
+
+        try {
+            getClient(adminToken).perform(post("/api/versioning/versions")
+                                 .param("summary", "check first version")
+                                 .contentType(MediaType.parseMediaType(RestMediaTypes.TEXT_URI_LIST_VALUE))
+                                 .content("/api/core/items/" + item.getID()))
+                                 .andExpect(status().isCreated())
+                                 .andExpect(jsonPath("$", Matchers.allOf(
+                                            hasJsonPath("$.version", is(3)),
+                                            hasJsonPath("$.summary", is("check first version")),
+                                            hasJsonPath("$.submitterName", is("first (admin) last (admin)")),
+                                            hasJsonPath("$.type", is("version"))
+                                            )));
+        } finally {
+            VersionBuilder.delete(idRef.get());
```

Comment:
This doesn't appear to be deleting the version you just POSTed.  It looks like it's deleting the version you created via VersionBuilder, unless I'm misreading the code.  I think this last POST needs to have an `andDo` clause to update the idRef.